src/selection.js

```diff
@@ -14,33 +14,72 @@ module.exports = Selection;
  * facilities to manipulate, save, restore the
  * selection, across DOM modifications.
  */
-function Selection(nativeSelection) {
+function Selection(nativeSelection, editor) {
```

Comment:
It's preferable to void circular dependencies, so these things are more usable by themselves.

In this case the editor is only needed to get its current elem.  The current elem can change though, so you do need the function.

Perhaps a good (and sort of future-proof) approach here is to just document the interface required: an "editor" needs to have a "currentElem" method, that's all (so people can use something simpler).  This also leaves the option to add more supporter methods/properties if necessary (optional ones, if backwards compatibility is to be maintained).